packages/venia-concept/src/components/ResponsiveImage/ResponsiveImage.js

```diff
@@ -0,0 +1,166 @@
+/**
+ * Autogenerate an image tag with `sizes` and `srcSet` attributes, using simple
+ * media conditions, a source URL, and an array of pixel widths for resized
+ * images to create. Renders the image resize URLs expected by venia-upward.
+ *
+ * Optionally, takes a render callback so you can use other DOM structures
+ * besides literal img tags to display responsive images. The img tag is
+ * required, however.
+ */
+
+import React, { Component } from 'react';
+import { arrayOf, func, number, oneOf, string } from 'prop-types';
+import makeMediaUrl from 'src/util/makeMediaUrl';
+
+class ResponsiveImage extends Component {
```

Comment:
I'm not sure this component is working correctly from looking at https://magento-venia-etnammagpk.now.sh/

It looks like the inline style image on the surrounding `span` is always overriding the `img`. It does't seem to resize (presumably as it's the low-res preview?) but the underlying `img` also doesn't look correctly sized.

For example, disable the `opacity: 0` in DevTools and it should show.
Before:
![screenshot 2019-01-15 at 16 36 28](https://user-images.githubusercontent.com/108052/51194966-08289c00-18e4-11e9-919d-c92d3fb4b5b5.png)

Reply:
After:
![screenshot 2019-01-15 at 16 38 22](https://user-images.githubusercontent.com/108052/51195005-22fb1080-18e4-11e9-915a-99e0280c53c8.png)

Reply:
Yes, that needs some work. With the new art direction capabilities of `srcset` and `<picture>`, I'm not sure we need to use CSS backgrounds here.

I left them in because I wanted to surgically refactor image tags rather than whole components if I could help it--plus, this type of usage forced me to think harder about the API of the ResponsiveImage tag. However, it may not be necessary to deviate from the sensible default of image + srcset, so if you think I should, I can simplify this code and remove the render prop feature.

I also observed while trying to demo this that resizing the viewport doesn't seem to trigger a dynamic update to the `currentSrc` of the image. I haven't gone and StackOverflown that yet, but I was wondering if that was Chrome's behavior on purpose. Maybe it thinks that img tags semantically should use only one source per the lifetime of the element, to conserve bandwidth. Any ideas, @rowan-m ? (I'll update this later if I learn this on my own through independent research).

Reply:
I think for `srset` the browser will apply the largest image it has in cache rather than request a smaller image during resize (even though it more accurately fits the media query). The images are assumed to be equivalent content, and the browser makes the swap call.

Whereas for `<picture>` it will swap when told because that's semi-intended for art direction (i.e. different content per media query) and the browser must swap to maintain the correct presentation.